### DISCUSSION TITLE
Is it possible to apply `visibility = "uncounted"` to an entire section?

### DISCUSSION THREAD

**Discussion by etiennebacher** [quarto-dev/quarto-cli#4798] [Q&A] [answered]
I'd like to prevent the slides in my "Appendix" section to appear in the progress bar and slides counter. I can use `{visibility="uncounted"}` as described [in the docs](https://quarto.org/docs/presentations/revealjs/advanced.html#uncounted-slides). However, it quickly becomes tedious if I have more than a few slides in the appendix. Is there a way to apply `visibility="uncounted"` to an entire section?

So far, I have:
```markdown
---
format: 
  revealjs:
    slide-number: true
---

Slide with main content

# Appendix {.appendix visibility="uncounted"}

---

## {visibility="uncounted"}

Slide with appendix content

---

## {visibility="uncounted"}

Another slide with appendix content
````

**Comment by cderv** [comment, marked as answer]
@etiennebacher as often with this attributes related question, you can write a simple Lua filter to help you with this 

````markdown
---
format: 
  revealjs:
    slide-number: true
    filters: [custom.lua]
---

Slide with main content

# Appendix {.appendix}

##

Slide with appendix content

##

Another slide with appendix content
````

`custom.lua`
````lua
local in_appendix = false

Header = function(h)
  if h.level == 1 then
    if h.classes:includes("appendix") then
      in_appendix = true
      h.attributes["visibility"] = "uncounted"
    else
      in_appendix = false
    end
  end
  if h.level == 2 and in_appendix then
    h.attributes["visibility"] = "uncounted"
  end
  return h
end
````

Though maybe when set to a h1 header (section title slide) we should apply to all the content... not sure if this is an expected default or not 🤔 - This is a revealjs feature though, https://revealjs.com/slide-visibility/#uncounted-slides so maybe it should be upstream...

**Reply by etiennebacher** [reply]
Thank you!
